client/verta/verta/_cli/deployment/update.py

```diff
@@ -18,16 +18,22 @@ def update():
 @click.argument("path", nargs=1, required=True)
 @click.option("--run-id", "-r", help="Experiment Run to deploy. Cannot be used alongside --model-version-id.")
 @click.option("--model-version-id", "-m", help="Model Version to deploy. Cannot be used alongside --run-id.")
+@click.option("--filename", "-f", help="Path to JSON or YAML config file. Can only be used alongside --workspace.")
 @click.option("--strategy", "-s", type=click.Choice(['direct', 'canary'], case_sensitive=False), help="Strategy to use to roll out new deployment.")
 @click.option("--canary-rule", "-c", multiple=True, help="Rule to use for canary deployment. Can only be used alongside --strategy=canary.")
 @click.option("--canary-interval", "-i", type=click.IntRange(min=0), help="Rollout interval, in seconds. Can only be used alongside --strategy=canary.")
 @click.option("--canary-step", type=click.FloatRange(min=0.0, max=1.0), help="Ratio of deployment to roll out per interval. Can only be used alongside --strategy=canary.")
 @click.option("--env-vars", type=str, help="Environment variables to set for the model build. The format is --env-vars '{\"VERTA_HOST\": \"app.verta.ai\"}'.")
 @click.option("--workspace", "-w", help="Workspace to use.")
 # TODO: more options
-def update_endpoint(path, run_id, model_version_id, strategy, canary_rule, canary_interval, canary_step, env_vars, workspace):
+def update_endpoint(path, run_id, model_version_id, filename, strategy, canary_rule, canary_interval, canary_step, env_vars, workspace):
     """Update an endpoint.
     """
+    non_file_options = (run_id, model_version_id, strategy, canary_rule, canary_interval, canary_step, env_vars)
+
+    if filename and any(non_file_options):
+        raise click.BadParameter("--filename can only be used alongside --workspace.")
```

Comment:
This wording sounds like "`--filename` can only be used if `--workspace` is _also_ used.
Maybe something like "--filename can't be used alongside other options except for --workspace"?

Reply:
Fixed.